learning_resources/serializers.py

```diff
@@ -438,3 +468,90 @@ class Meta:
             "resource_readable_num",
             "resource_type",
         ]
+
+
+class UserListSerializer(serializers.ModelSerializer, WriteableTopicsMixin):
+    """
+    Simplified serializer for UserList model.
+    """
+
+    item_count = serializers.SerializerMethodField()
+
+    def get_item_count(self, instance) -> int:
+        """Return the number of items in the list"""
+        return getattr(instance, "item_count", None) or instance.resources.count()
```

Comment:
Question: When would the first condition be used vs the second?

I would have thought it would also use `instance.resources.count()` since the model doesn't have a. `item_count` field/attribute.

Reply:
`item_count` is added via the view queryset:
```python
    def get_queryset(self):
        """Return a queryset for this user"""
        return (
            UserList.objects.all()
            .prefetch_related("author", "topics")
            .annotate(item_count=Count("children"))
        )
```
Copied over from `course_catalog`, I believe it may be [marginally more efficient](https://stackoverflow.com/questions/7957569/in-django-is-aggregatecount-faster-or-better-than-count-in-anyway) to run `annotate(item_count=Count("children")` than `instance.resources.count()` - but the latter is there as a backup in case the serializer is used outside of a view.